core/examples/http_bench.js

```diff
@@ -1,11 +1,77 @@
 // This is not a real HTTP server. We read blindly one time into 'requestBuf',
 // then write this fixed 'responseBuf'. The point of this benchmark is to
 // exercise the event loop in a simple yet semi-realistic way.
-const OP_LISTEN = 1;
-const OP_ACCEPT = 2;
-const OP_READ = 3;
-const OP_WRITE = 4;
-const OP_CLOSE = 5;
+const registry = {};
+
+function initOps(opsMap) {
+  for (const [name, opId] of Object.entries(opsMap)) {
+    const op = registry[name];
+
+    if (!op) {
+      continue;
+    }
+
+    op.setOpId(opId);
+  }
+}
+
+class Op {
+  constructor(name) {
+    if (typeof registry[name] !== "undefined") {
+      throw new Error(`Duplicate op: ${name}`);
+    }
+
+    this.name = name;
+    this.opId = 0;
+    registry[name] = this;
+  }
+
+  setOpId(opId) {
+    this.opId = opId;
+  }
+
+  static handleAsyncMsgFromRust(opId, buf) {
+    const record = recordFromBuf(buf);
+    const { promiseId } = record;
+    const p = promiseMap.get(promiseId);
+    promiseMap.delete(promiseId);
+    p.resolve(record);
+  }
+
+  static sendSync(opId, arg, zeroCopy) {
+    const buf = send(0, opId, arg, zeroCopy);
+    return recordFromBuf(buf);
+  }
+
+  static sendAsync(opId, arg, zeroCopy = null) {
+    const promiseId = nextPromiseId++;
+    const p = createResolvable();
+    promiseMap.set(promiseId, p);
+    send(promiseId, opId, arg, zeroCopy);
+    return p;
+  }
+}
+
+class HttpOp extends Op {
+  /** Returns i32 number */
+  sendSync(arg, zeroCopy = null) {
+    const res = HttpOp.sendSync(this.opId, arg, zeroCopy);
+    return res.result;
+  }
+
+  /** Returns Promise<number> */
+  async sendAsync(arg, zeroCopy = null) {
+    const res = await HttpOp.sendAsync(this.opId, arg, zeroCopy);
+    return res.result;
+  }
+}
+
+const OP_LISTEN = new HttpOp("listen");
+const OP_ACCEPT = new HttpOp("accept");
+const OP_READ = new HttpOp("read");
+const OP_WRITE = new HttpOp("write");
+const OP_CLOSE = new HttpOp("close");
```

Comment:
It's not clear to me if op ids have been assigned at this point? It seems not?

Reply:
No, they're not. Op ids are assigned after explicitly getting them from Rust with `Deno.core.getOps()`.